psi4/src/psi4/libpe/psipe.cc

```diff
@@ -0,0 +1,259 @@
+    mult.push_back(std::make_shared<Matrix>("AO EFP Quadrupole ZZ", basisset_->nbf(), basisset_->nbf()));
+    mult.push_back(std::make_shared<Matrix>("AO EFP Quadrupole XY", basisset_->nbf(), basisset_->nbf()));
+    mult.push_back(std::make_shared<Matrix>("AO EFP Quadrupole XZ", basisset_->nbf(), basisset_->nbf()));
+    mult.push_back(std::make_shared<Matrix>("AO EFP Quadrupole YZ", basisset_->nbf(), basisset_->nbf()));
+    mult.push_back(std::make_shared<Matrix>("AO EFP Octupole XXX", basisset_->nbf(), basisset_->nbf()));
+    mult.push_back(std::make_shared<Matrix>("AO EFP Octupole YYY", basisset_->nbf(), basisset_->nbf()));
+    mult.push_back(std::make_shared<Matrix>("AO EFP Octupole ZZZ", basisset_->nbf(), basisset_->nbf()));
+    mult.push_back(std::make_shared<Matrix>("AO EFP Octupole XXY", basisset_->nbf(), basisset_->nbf()));
+    mult.push_back(std::make_shared<Matrix>("AO EFP Octupole XXZ", basisset_->nbf(), basisset_->nbf()));
+    mult.push_back(std::make_shared<Matrix>("AO EFP Octupole XYY", basisset_->nbf(), basisset_->nbf()));
+    mult.push_back(std::make_shared<Matrix>("AO EFP Octupole YYZ", basisset_->nbf(), basisset_->nbf()));
+    mult.push_back(std::make_shared<Matrix>("AO EFP Octupole XZZ", basisset_->nbf(), basisset_->nbf()));
+    mult.push_back(std::make_shared<Matrix>("AO EFP Octupole YZZ", basisset_->nbf(), basisset_->nbf()));
+    mult.push_back(std::make_shared<Matrix>("AO EFP Octupole XYZ", basisset_->nbf(), basisset_->nbf()));
+    multipole_integrals->set_origin(site);
+	multipole_integrals->compute(mult);
+    
+    auto res = std::make_shared<Matrix>("result", basisset_->nbf(), basisset_->nbf());
+    
+    int total_components = 0;
+    for (int i = 0; i <= order; ++i) {
+        total_components += libcppe::multipole_components(i);
+    }
+    
+    for (int l = 0; l < total_components; ++l) {
+        // std::cout << "---" << std::setprecision(12) << std::endl;
+        // std::cout << "L: " << l << std::endl;
+        // std::cout << "coeff = " << mult_coeffs[l] << " moment = " << moments[from_alphabetical_to_psi4[l]] << std::endl;
+        mult[l]->scale(-1.0 * mult_coeffs[l] * moments[from_alphabetical_to_psi4[l]]);
+        res->add(mult[l]);
+    }
+    
+    return res;
+}
+
+SharedMatrix PeIntegralHelper::compute_field_integrals(Vector3 site, arma::vec moment) {
+    std::vector<SharedMatrix> fields;
+    fields.push_back(std::make_shared<Matrix>("AO Field X", basisset_->nbf(), basisset_->nbf()));
+    fields.push_back(std::make_shared<Matrix>("AO Field Y", basisset_->nbf(), basisset_->nbf()));
+    fields.push_back(std::make_shared<Matrix>("AO Field Z", basisset_->nbf(), basisset_->nbf()));
+    
+    auto integrals = std::make_shared<IntegralFactory>(basisset_, basisset_, basisset_, basisset_);
+    std::shared_ptr<OneBodyAOInt> field_integrals_ = static_cast<std::shared_ptr<OneBodyAOInt>>(integrals->electric_field());
```

Comment:
I think `auto field_integrals_ = std::make_shared<OneBodyAOInt>(integrals->electric_field());` should suffice here too.

Reply:
Resolved in b9742ad4d.